openni2_camera/src/openni2_driver.cpp

```diff
@@ -849,6 +855,9 @@ void OpenNI2Driver::initDevice()
     }
     catch (const OpenNI2Exception& exception)
     {
+      std_msgs::String msg;
```

Comment:
Is this also related to a failure of getting camera serial? Because we also make save an "camera serial failed" event in this case.

Reply:
The "get camera serial failure" is actually a "device open failure" under the hood, that is why I called it as such and also add the capture here. See https://intermodalics.slack.com/archives/C3S48Q0E9/p1637596693277600 for how the error pops up through different code paths.